apisix/plugins/key-auth.lua

```diff
@@ -110,6 +114,12 @@ function _M.rewrite(conf, ctx)
     end
     core.log.info("consumer: ", core.json.delay_encode(consumer))
 
+    if conf.hide_credentials then
+        core.request.set_header(ctx, conf.header, nil)
+        core.request.get_uri_args(ctx)
+        core.request.set_uri_args(ctx, {})
```

Comment:
What is the purpose of doing this?

Reply:
Hide information from query

Reply:
I think we don't necessary to remove the all query params
If the key is in the header, we only need to delete the key in the headers.
If the key is in the query, we only need to delete the key in the query params.

cc @spacewander right?

Reply:
Yes, we should only remove the key.

Reply:
OK, I'll revise it right away.